structopt-derive/src/lib.rs

```diff
@@ -516,19 +544,11 @@ fn gen_augment_clap(fields: &[Field]) -> quote::Tokens {
 }
 
 fn gen_clap_enum(enum_attrs: &[Attribute]) -> quote::Tokens {
-    let enum_attrs: Vec<_> = extract_attrs(enum_attrs, AttrSource::Struct).collect();
-    let name = from_attr_or_env(&enum_attrs, "name", "CARGO_PKG_NAME");
-    let version = from_attr_or_env(&enum_attrs, "version", "CARGO_PKG_VERSION");
-    let author = format_author(from_attr_or_env(&enum_attrs, "author", "CARGO_PKG_AUTHORS"));
-    let about = from_attr_or_env(&enum_attrs, "about", "CARGO_PKG_DESCRIPTION");
-
+    let gen = gen_clap(enum_attrs);
     quote! {
         fn clap<'a, 'b>() -> _structopt::clap::App<'a, 'b> {
-            let app = _structopt::clap::App::new(#name)
-                .version(#version)
-                .author(#author)
-                .about(#about)
-                .setting(_structopt::clap::AppSettings::SubcommandRequired);
+            let app = #gen
+                .setting(_structopt::clap::AppSettings::SubcommandRequiredElseHelp);
```

Comment:
That's OK for me. @williamyaoh OK?

Reply:
Seems like it makes `structopt` more usable. No qualms here.